agent-inject/agent/agent.go

```diff
@@ -504,7 +513,7 @@ func (a *Agent) Validate() error {
 	}
 
 	if a.ConfigMapName == "" {
-		if a.Vault.Role == "" {
+		if a.Vault.Role == "" && a.Vault.AuthType == DefaultVaultAuthType {
```

Comment:
I think this isn't specific to K8s auth method, but any auth method would specify a role.

Reply:
My reasoning on this was - because vault-k8s currently maps the annotation `vault.hashicorp.com/role` directly to the field:

```
{
  "auto_auth": {
    "method": {
      "config": {
        "role": "<value of the above annotation>"
```

...but the TLS certs auth method for example doesn't have a field `role`, but rather one called `name` - I put this check in to preserve backwards compatibility with how things work now.

Essentially, the following two are equivalent:
- `vault.hashicorp.com/role`  <-- specific to the field "role"
- `vault.hashicorp.com/auth-config-role`  <-- "role" could be replaced with any parameter name (and have multiple specified), to support auth methods that may take differently named parameters

Reply:
Thanks @pbar1, I forgot that cert auto-auth was slightly different.